pex/resolve/lockfile/download_manager.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import
+
+import os
+
+from pex.common import atomic_directory
+from pex.resolve.locked_resolve import Artifact
+from pex.typing import TYPE_CHECKING
+from pex.variables import ENV
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+    import attr  # vendor:skip
+else:
+    from pex.third_party import attr
+
+
+@attr.s(frozen=True)
+class DownloadedArtifact(object):
+    path = attr.ib()  # type: str
+
+    def fingerprint(self):
+        # type: () -> str
+        with open(os.path.join(os.path.dirname(self.path), "sha1"), "rb") as fp:
```

Comment:
Generally, should we use sha256 in pex for more robust security? sha1 is susceptible to practical collision attacks, and it looks like we use it a lot.

Reply:
Yeah, so we're exactly as vulnerable as git. I really don't see the practical concern on the collision end as a result; i.e.: people have much bigger problems if this is real since they ~all use git!

On the practical side of this is filesystem path name length limits. This is already a problem we run into and pex has warnings it emits when things are looking dangerous. Going to sha256 would net 24 more characters in some paths and 48 in others (venvs).

Reply:
This may be worthwhile to investigate; although using a lock file mitigates afaict since the primary cache key becomes the hash algorithm used by the artifact - generally sha256. All the rest of the keys are then just internal. To attack one of those you'd need to 1st break the artifact sha256 hash. So, for those _not_ using lock files, Pex is more vulnerable than it need be as you say. The path lengths are the sticky wicket.

Reply:
https://github.com/pantsbuild/pex/issues/1655